postreise/plot/plot_sim_vs_hist.py

```diff
@@ -10,41 +11,66 @@ class PlotSettings:
     size_inches = (20, 10)
 
 
-def plot_generation_sim_vs_hist(sim_gen, hist_gen, s_info, state, showmax=True):
-    """
-    Plot simulated vs historical generation of each resource in the scenario
+def plot_generation_sim_vs_hist(
+    scenario,
+    hist_gen,
+    state,
+    show_max=True,
+    plot_show=True,
+):
+    """Plot simulated vs historical generation of each resource in the scenario
     for the given state. Optionally include max generation.
 
-    :param pandas.DataFrame sim_gen: simulated generation dataframe
-    :param pandas.DataFrame hist_gen: historical generation dataframe
-    :param powersimdata.design.ScenarioInfo s_info: scenario info instance.
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance.
+    :param pandas.DataFrame hist_gen: historical generation data frame. Columns are
+        resources and indices are state names.
     :param str state: the US state
-    :param bool showmax: determine whether to additionally plot max generation of each resource
+    :param bool show_max: determine whether to additionally plot max generation of each
+        resource.
+    :param plot_show: show the plot or not, defaults to True.
+    :return: (*matplotlib.axes.Axes*) -- axes object of the plot.
+    :raises TypeError:
+        if ``hist_gen`` is not a data frame.
+        if ``state`` is not a str.
+    :raises ValueError: if ``state is not in ``hist_gen`` or ``scenario``.
     """
-    labels = list(sim_gen.columns)
-    all_resources = s_info.get_available_resource("all")
+    if not isinstance(hist_gen, pd.DataFrame):
+        raise TypeError("hist_gen must be a pandas.DataFrame")
+    if not isinstance(state, str):
+        raise TypeError("state must be a str")
+
+    sim_gen = sum_generation_by_state(scenario)
+    if state not in sim_gen.index or state not in hist_gen.index:
+        raise ValueError("Invalid state")
+
+    all_resources = list(sim_gen.columns)
 
     sim = [int(round(sim_gen.loc[state, res])) for res in all_resources]
     hist = [int(round(hist_gen.loc[state, res])) for res in all_resources]
 
-    def calc_max(gen_type):
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore")
-            return int(
-                s_info.get_capacity(gen_type, state) * len(s_info.pg.index) / 1000
-            )
+    def calc_max(fuel):
+        grid = scenario.get_grid()
+        loadzone = grid.model_immutables.area_to_loadzone(state, "state")  # noqa
+        capacity = grid.plant.query("type == @fuel & zone_name == @loadzone")[
```

Comment:
If `loadzone` is a set, do we need the equivalent of `plant["zone_name"].isin(loadzone)` instead of `zone_name == @loadzone`? Just looking at the implementation in `ScenarioInfo.get_capacity` which was used before.

Reply:
You are right. For the `query` to work, it needs to be a list

Reply:
```
>>> test = pd.DataFrame({"zone_name": ["A", "B", "C", "D", "E"], "type": ["hydro", "ng", "hydro", "wind", "hydro"]})
>>> fuel = "hydro"
>>> loadzone = {"A", "B", "C"}
>>> test.query("type == @fuel & zone_name == @loadzone")
mpty DataFrame
Columns: [zone_name, type]
Index: []
```
and
```
>>> loadzone = ["A", "B", "C"]
>>> test.query("type == @fuel & zone_name == @loadzone")
  zone_name   type
0         A  hydro
2         C  hydro
```

Reply:
Neat, I'm surprised a list works in that case, but it's convenient.

Reply:
I had the same discussion with @danielolsen this morning!

Reply:
`query` is powerful and spooky.

Reply:
`loadzone` is now a `list`

Reply:
TIL!